plugin/src/main/java/org/opensearch/ml/plugin/MachineLearningPlugin.java

```diff
@@ -842,8 +842,8 @@ public List<ExecutorBuilder<?>> getExecutorBuilders(Settings settings) {
         FixedExecutorBuilder executeThreadPool = new FixedExecutorBuilder(
             settings,
             EXECUTE_THREAD_POOL,
-            Math.max(1, OpenSearchExecutors.allocatedProcessors(settings) - 1),
-            10,
+            OpenSearchExecutors.allocatedProcessors(settings) * 4,
```

Comment:
The function allocateProcessors returns a int type. I think it is safer to use 
Math.max(4, OpenSearchExecutors.allocatedProcessors(settings) *4).

Reply:
got it, thanks